cpp/src/tsne/tsne_runner.cuh

```diff
@@ -78,6 +89,58 @@ class TSNE_runner {
       CUML_LOG_WARN(
         "# of Nearest Neighbors should be at least 3 * perplexity. Your results"
         " might be a bit strange...");
+
+    auto stream         = handle_.get_stream();
+    const value_idx dim = params.dim;
+
+    if (params.init == TSNE_INIT::RANDOM) {
+      random_vector(Y, -0.0001f, 0.0001f, n * dim, stream, params.random_state);
+    } else if (params.init == TSNE_INIT::PCA) {
+      rmm::device_uvector<float> components(p * dim, stream);
```

Comment:
Use `raft::device_vector` and `raft::device_matrix` instead of `rmm::device_uvector`. This will allow you to use API of Raft that use mdspan (`raft::device_matrix_view`, ...) in the functions below (`raft::stats::mean`, ...)